Better error when user tries to submit workflow without rabbitmq broker

When a user creates a profile with `verdi presto` without rabbitmq broker, and tries to submit a Workflow to a daemon via `submit` they will be greeted with this assert

```
    113     runner = manager.get_manager().get_runner()
    114     assert runner.persister is not None, 'runner does not have a persister'
--> 115     assert runner.controller is not None, 'runner does not have a controller'
    116 
    117     process_inited = instantiate_process(runner, process, **inputs)

AssertionError: runner does not have a controller
```

Since this will likely become a common problem for newcommers, we should provide a clear error message.

Should be done as a follow-up (or part of #6351)

--- a/src/aiida/engine/launch.py
+++ b/src/aiida/engine/launch.py
@@ -111,8 +111,16 @@ def submit(
         raise InvalidOperation('Cannot use top-level `submit` from within another process, use `self.submit` instead')
 
     runner = manager.get_manager().get_runner()
+
+    if runner.controller is None:
+        raise InvalidOperation(
+            'Cannot submit because the runner does not have a process controller, probably because the profile does '
+            'not define a broker like RabbitMQ. If a RabbitMQ server is available, the profile can be configured to '
+            'use it with `verdi profile configure-rabbitmq`. Otherwise, use :meth:`aiida.engine.launch.run` instead to '
+            'run the process in the local Python interpreter instead of submitting it to the daemon.'
+        )
+
     assert runner.persister is not None, 'runner does not have a persister'
-    assert runner.controller is not None, 'runner does not have a controller'
 
     process_inited = instantiate_process(runner, process, **inputs)
 

--- a/tests/engine/test_launch.py
+++ b/tests/engine/test_launch.py
@@ -20,6 +20,16 @@ from aiida.plugins import CalculationFactory
 ArithmeticAddCalculation = CalculationFactory('core.arithmetic.add')
 
 
+@pytest.fixture
+def arithmetic_add_builder(aiida_code_installed):
+    builder = ArithmeticAddCalculation.get_builder()
+    builder.code = aiida_code_installed(default_calc_job_plugin='core.arithmetic.add', filepath_executable='/bin/bash')
+    builder.x = orm.Int(1)
+    builder.y = orm.Int(1)
+    builder.metadata = {'options': {'resources': {'num_machines': 1, 'num_mpiprocs_per_machine': 1}}}
+    return builder
+
+
 @calcfunction
 def add(term_a, term_b):
     return term_a + term_b
@@ -69,16 +79,26 @@ class AddWorkChain(WorkChain):
 
 
 @pytest.mark.usefixtures('started_daemon_client')
-def test_submit_wait(aiida_code_installed):
+def test_submit_wait(arithmetic_add_builder):
     """Test the ``wait`` argument of :meth:`aiida.engine.launch.submit`."""
-    builder = ArithmeticAddCalculation.get_builder()
-    builder.code = aiida_code_installed(default_calc_job_plugin='core.arithmetic.add', filepath_executable='/bin/bash')
-    builder.x = orm.Int(1)
-    builder.y = orm.Int(1)
-    builder.metadata = {'options': {'resources': {'num_machines': 1, 'num_mpiprocs_per_machine': 1}}}
-    node = launch.submit(builder, wait=True, wait_interval=0.1)
+    node = launch.submit(arithmetic_add_builder, wait=True, wait_interval=0.1)
     assert node.is_finished, node.process_state
     assert node.is_finished_ok, node.exit_code
+
+
+def test_submit_no_broker(arithmetic_add_builder, monkeypatch, manager):
+    """Test that ``submit`` raises ``InvalidOperation`` if the runner does not have a controller.
+
+    The runner does not have a controller if the runner was not provided a communicator which is the case for profiles
+    that do not define a broker.
+    """
+    runner = manager.get_runner()
+    monkeypatch.setattr(runner, '_controller', None)
+
+    with pytest.raises(
+        exceptions.InvalidOperation, match=r'Cannot submit because the runner does not have a process controller.*'
+    ):
+        launch.submit(arithmetic_add_builder)
 
 
 def test_await_processes_invalid():

